lib/handler/redirect.c

```diff
@@ -101,6 +140,7 @@ h2o_redirect_handler_t *h2o_redirect_register(h2o_pathconf_t *pathconf, int inte
     self->super.on_req = on_req;
     self->internal = internal;
     self->status = status;
-    self->prefix = h2o_strdup(NULL, prefix, SIZE_MAX);
+    build_prefix_list(prefix, &self->prefix_list);
```

Comment:
Maybe change the signature of this function to `self->prefix_list = build_prefix_list(prefix)` now that we have refactored the design?

Such change would make it easier to write unit tests, and I might even envisage changing the function to `h2o_split` that takes the split char as the argument, and moving it into string.c. Not that we need to do now, but...